docs/compaction-merge-loops.md

```diff
@@ -0,0 +1,153 @@
+# Compaction Merge Loops
+
+Compaction operations transform a set of input kvsets into a set of
+new output kvsets.  They are used are used to manage CN Tree shape
+(fanout, node size, node length), and to control space amp (e.g., by
+eliminating deleted key-value pairs).
+
+There are three types of compaction operations:
+- spill: consume one or more kvsets from the root node, produce at most
+  one output kvset for each leaf node.
+- kv-compact: consume two or more kvsets, produce one kvset
+- k-compact: similar to kv-compact, but without compacting vblocks
+
+HSE employs two additional operations that are an integral part of CN
+Tree maintenance but strictly speaking are not compaction operations:
+- split: split one kvset into two kvsets
+- join: join two kvsets into one kvset
+
+While split and join do transform input kvsets into a set of new
+output kvsets, they are meta-data-only operations.  They don't rewrite
+kblocks or vblocks and they don't use compaction merge loops.  This
+document does not apply to split or join operations.
+
+Spill, k-compact and kv-compact operations follow these steps:
+```
+  1. read entries from one or more input kvsets
```

Comment:
Take the list out of the formatted block.

Markdown is nice, and let's you do this:

```
1.
1.
1.
```
which comes out to

1.
2.
3.

Otherwise any time you add something, it requires changing all the lines

Reply:
I wanted to make a figure out of it so I put all of in a  ```text section.